### PR TITLE
update component api docs

### PR DESCRIPTION
**Pull request by lili21**
#4601

### CODE REVIEW

File: docs/docs/ref-02-component-api.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Instances of a React Component are created internally in React when rendering. T
 ```javascript
 setState(function|object nextState[, function callback])
 ```
-Merges nextState with the current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.
+Merges (shallow merge) nextState with the current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.
```
Comment (jimfb) [inline docs/docs/ref-02-component-api.md]: I feel like this reads a little awkwardly.  Maybe "Merges (shallowly, not recursively) nextState with the current state" or "Merges nextState with the current state (shallowly, not recursively)".  Or reword so it flows more naturally with the paragraph text.  The phrase "Merges (shallow Merge) nextState" just feels awkward; the rewording has the added bonus of clearifying what shallow means for anyone unfamiliar with the term (intro-to-react api).  That said, I don't have a strong objection here.  @zpao or @spicyj for a final decision here.

Comment (yaycmyk) [inline docs/docs/ref-02-component-api.md]: or "non-recursively"

Comment (zpao) [inline docs/docs/ref-02-component-api.md]: Maybe this?

"Performs a shallow merge of `nextState` into the current state."

Comment (jimfb) [inline docs/docs/ref-02-component-api.md]: Yep, let's go with that.

**Comment by lili21** [comment]
Already update the content.

**Comment by zpao** [comment]
:thumbsup: thanks!

**Pull request merged**